torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -36,7 +52,29 @@ void LinearImpl::pretty_print(std::ostream& stream) const {
 
 Tensor LinearImpl::forward(const Tensor& input) {
   AT_ASSERT(!options.with_bias_ || bias.defined());
-  return torch::linear(input, weight, bias);
+  auto input_view = input.view_as(input);
```

Comment:
Why is this necessary? Sorry if I missed something obvious.

Reply:
This is not required. If the cost of creating a view is a concern, i will edit the code to avoid this. 

I was assuming that view adds negligible overhead, and hence creating a view upfront for less LoC.